### PR TITLE
Adding a LinearGradient subclass of ColorPalette

### PR DESCRIPTION
**Pull request by wolfposd**
Adds a Subclass of ColorPalette, which enables to specify GradientLike colorpalettes.

new LinearGradientColorPalette(Color.RED, 18, Color.Blue);
// creates a ColorPalette with colors ranging between Red and Blue, with 18 colors in between

new LinearGradientColorPalette(Color.RED, 5, Color.BLUE, 5, Color.GREEN); 
// creates a ColorPalette with colors ranging from Red over Blue to Green, with 5 colors in between each

### CODE REVIEW

File: src/main/java/wordcloud/palette/LinearGradientColorPalette.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A LinearGradient ColorPalette
+ * 
+ * @author &#64;wolfposd
+ */
+public class LinearGradientColorPalette extends ColorPalette
+{
```
Comment (kennycason) [inline src/main/java/wordcloud/palette/LinearGradientColorPalette.java]: could you move the parentheses to follow the format 
```java
public class LinearGradientColorPalette extends ColorPalette {
```

Comment (kennycason) [inline src/main/java/wordcloud/palette/LinearGradientColorPalette.java]: the same note follows all the open { in the code.

File: src/main/java/wordcloud/palette/LinearGradientColorPalette.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     * @param color2
+     *            middle color
+     * @param betweenC2andC3
+     *            number of colors to be generated between color2 and color3
+     * @param color3
+     *            end color
+     */
+    public LinearGradientColorPalette(Color color1, int betweenC1andC2, Color color2, int betweenC2andC3, Color color3)
+    {
+        ArrayList<Color> cs = new ArrayList<>();
```
Comment (kennycason) [inline src/main/java/wordcloud/palette/LinearGradientColorPalette.java]: prefer `List<Color> cs = new ArrayList<>();`

File: src/main/java/wordcloud/palette/LinearGradientColorPalette.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+        for (Color c : createLinearGradient(color1, color2, betweenC1andC2))
+        {
+            cs.add(c);
+        }
+
+        for (Color c : createLinearGradient(color2, color3, betweenC2andC3))
+        {
+            if (c.equals(color2)) // already being added by first gradient
```
Comment (kennycason) [inline src/main/java/wordcloud/palette/LinearGradientColorPalette.java]: even for one line if statements I prefer to wrap in { }
Also for trivial if statements (return/break/continues) i don't mine just doing it in one line. 
```java
if (c.equals.color2) { continue; }
```

File: src/main/java/wordcloud/palette/LinearGradientColorPalette.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+     *            start color
+     * @param color2
+     *            end color
+     * @param amountBetweenC1andC2
+     *            specifies the amount of colors in this gradient between color1 and color2
+     * @return List of colors in this gradient
+     */
+    public static List<Color> createLinearGradient(Color color1, Color color2, int amountBetweenC1andC2)
+    {
+        ArrayList<Color> cols = new ArrayList<>();
```
Comment (kennycason) [inline src/main/java/wordcloud/palette/LinearGradientColorPalette.java]: same comment as above.

**Comment by kennycason** [comment]
As a general note I try to use final as much as possible as well. :)

**Comment by kennycason** [comment]
Thanks a lot for the pull request, I like this feature. I had a few style recommendations but after that I'll happily merge it in!

**Comment by kennycason** [comment]
If you could update the README with the different color palette options and maybe generate a sample wordcloud and stick along with the other samples that would be great. Then we can show it off

**Comment by wolfposd** [comment]
> As a general note I try to use final as much as possible as well. :)
this makes inheritance kind of awkward

Do you by chance have a textfile which is suitable for nice wordclouds?

**Comment by kennycason** [comment]
> Do you by chance have a textfile which is suitable for nice wordclouds?
I have a few resource text files, but I don't know how "good" they are. I can check the resources folder later.

>this makes inheritance kind of awkward
I don't mean making classes and method final. I'm also not referring to the protected variable you made. 
I'm referring to method parameters and all local variables. Don't worry about it for this PR. I need to give my code another glance and make sure I adhere to my own standards :)

**Comment by kennycason** [comment]
Thanks a lot for the PR :+1:

**Comment by kennycason** [comment]
I'll do some local testing, and have one more branch to merge, then i'll up the version and deploy!

**Pull request merged**